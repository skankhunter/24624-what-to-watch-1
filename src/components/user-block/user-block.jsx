import React, {PureComponent} from "react";
import {Link} from "react-router-dom";
import {withRouter} from "react-router";
import {connect} from "react-redux";
import {compose} from "redux";
import PropTypes from "prop-types";

class UserBlock extends PureComponent {
  constructor(props) {
    super(props);

    this._handelAvatarClick = this._handelAvatarClick.bind(this);
  }

  render() {
    const {authorized, user} = this.props;

    if (!authorized) {
      return (
        <div className="user-block">
          <Link to="/login" className="user-block__link">
            Sign in
          </Link>
        </div>
      );
    } else {
      return (
        <div className="user-block">
          <div className="user-block__avatar" onClick={this._handelAvatarClick}>
            <img
              src={`https://es31-server.appspot.com/${user.userAvatar}`}
              alt={user.userName}
              width="63"
              height="63"
            />
          </div>
        </div>
      );
    }
  }

  _handelAvatarClick() {
    const {history} = this.props;

    history.push(`/favorites`);
  }
}

const mapStateToProps = (state) => ({
  user: state.user.currentUser,
  authorized: state.user.authorized
});

UserBlock.propTypes = {
  authorized: PropTypes.bool.isRequired,
  user: PropTypes.object.isRequired,
  history: PropTypes.object.isRequired,
};

export {UserBlock};

export default compose(
    connect(
        mapStateToProps,
        null
    ),
    withRouter
)(UserBlock);
