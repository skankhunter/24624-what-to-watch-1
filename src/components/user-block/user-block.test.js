import React from "react";
import renderer from "react-test-renderer";

import {UserBlock} from "./user-block.jsx";

const mocks = {
  authorized: true,
  user: {
    userId: 1,
    userEmail: `mail`,
    userName: `name`,
    userAvatar: `ava`
  },
  history: {
    push: jest.fn()
  }
};

describe(`UserBlock:`, () => {
  it(`Correctly renders after relaunch`, () => {
    const tree = renderer.create(<UserBlock {...mocks} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
