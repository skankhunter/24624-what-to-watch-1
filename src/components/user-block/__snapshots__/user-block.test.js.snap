// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserBlock: Correctly renders after relaunch 1`] = `
<div
  className="user-block"
>
  <div
    className="user-block__avatar"
    onClick={[Function]}
  >
    <img
      alt="name"
      height="63"
      src="https://es31-server.appspot.com/ava"
      width="63"
    />
  </div>
</div>
`;
